dandi/utils.py

```diff
@@ -890,3 +891,66 @@ def joinurl(base: str, path: str) -> str:
         return path
     else:
         return base.rstrip("/") + "/" + path.lstrip("/")
+
+
+def get_retry_after(response: requests.Response) -> Optional[int]:
+    """If provided and parsed ok, returns duration in seconds to sleep before retry.
+
+    If not provided in the response header `Retry-After`, would
+    return None.
+    If parsing fails, or provided date/sleep does not make sense
+    since either too far in the past (over 2 seconds) or in the future
+    (over a week), would return None.
+    """
+    if_unparsable = None
+    retry_after = response.headers.get("Retry-After")
+    if retry_after is None:
+        return None
+    sleep_amount: int | None
+    if retry_after.isdecimal():
+        sleep_amount = int(retry_after)
+    else:
+        # else if it is a datestamp like "Wed, 21 Oct 2015 07:28:00 GMT"
+        # we could parse it and calculate how long to sleep
+        current_date = datetime.datetime.now(datetime.timezone.utc)
+        try:
+            retry_after_date = parsedate_to_datetime(retry_after)
+        except (ValueError, TypeError) as exc_ve:
+            # our code or response is wrong, do not crash but issue warning
+            # and continue with "if_unparsable" sleep logic
+            lgr.warning(
+                "response %d has incorrect date in Retry-After=%r: %s. " "Returning %r",
+                response.status_code,
+                retry_after,
+                exc_ve,
+                if_unparsable,
+            )
+            sleep_amount = if_unparsable
+        else:
+            difference = retry_after_date - current_date
+            sleep_amount = int(difference.total_seconds())
+
+        if sleep_amount:
```

Comment:
Shouldn't this block be dedented so that integer Retry-Afters also get checked against a week?

Reply:
yes! will do